src/Icicle/Avalanche/Statement/Simp.hs

```diff
@@ -116,7 +116,7 @@ convertValues a_fresh statements
            -> XPrim a_fresh (PrimUnsafe (PrimUnsafeArrayCreate t))
                `xApp` n
          (x : xs)
-           -> XPrim a_fresh (PrimUpdate (PrimUpdateArrayPut t))
+           -> XPrim a_fresh (PrimArray  (PrimArrayPutImmutable t))
```

Comment:
I think this can be a mutable put, since it is operating on a newly created array that nothing else can reference. also, I think this transform is done after mutate, so won't be recovered by mutate.
(not that we have many array literals)